### ISSUE TITLE
Singleton support

### ISSUE THREAD

**Issue by dineshbvadhia** [chicago-cdac/lmdb-dict#1]
Hello. In an application with multiple modules, which some or all need access to the lmdb db, how is this achieved with lmdb-dict? Typically, this is achieved with a parametric singleton.

**Comment by jesteria** [comment]
Hello!

1) I might not understand your particular use-case. But I would say that in general in Python, if it's important to you to retain a reference to a specific instance of a class across multiple modules, this can be achieved easily.

    Given a package of the following modules …

    - db.py
    - aaa.py
    - bbb.py

    … you might instantiate `data = MyDatabase()` at the top level of module `db`, and import that instance in both modules `aaa` and `bbb` as `from .db import data`.

    That is, in this case, there is no need for the library to supply a singleton-producing interface. It is up to the library user to instantiate and use as many or as few instances of the class as they like.

2) In fact there _is_ need for something like a singleton pattern in `lmdb_dict`, but this is an implementation detail and needn't concern the user. As [mentioned in the readme](https://github.com/uchicago-dsi/lmdb-dict#general-use), and [handled in the code](https://github.com/uchicago-dsi/lmdb-dict/blob/39f4213a3634f3b8060725f2bbdeb171be284332/lmdb_dict/mapping/abc.py#L139), it _would_ be a problem for the library to open multiple concurrent connections to the same database, and it would be a problem to maintain multiple concurrent serialization caches. And so, behind the scenes, the library does ensure that duplicative instances of the class which refer to the same database and/or cache receive the same database connection and/or cache. As such, equal instances of the class should operate identically and without error. In the above example, you _could_ simply instantiate `data = MyDatabase()` in both modules `aaa` and `bbb` – without any shared module `db`. The important bits will operate as singletons. But the user requested two instances of the interface class, and receives two instances. 

In short, it should be up to you and your code how instances of the database class are managed. And the library should protect your code from error.

Hope that helps!
